### PR TITLE
Fix/compile twice foundry

### PR DESCRIPTION
**Pull request by portdeveloper**
### This PR:
-> Removes the build target in the makefile
-> 
-> 

## Additional Information

- [x] I have read the [contributing docs](/scaffold-eth/scaffold-eth-2/blob/main/CONTRIBUTING.md) (if this is your first contribution)
- [x] This is not a duplicate of any [existing pull request](https://github.com/scaffold-eth/scaffold-eth-2/pulls)

## Related Issues

_Closes #1019

### CODE REVIEW

File: packages/foundry/Makefile [hunk 1]
```diff
@@ -100,4 +95,5 @@ lint:
 verify:
 	forge script script/VerifyAll.s.sol --ffi --rpc-url $(RPC_URL)
 
-build-and-verify: build verify
+# Verify and generate ABIs
+verify-and-generate-abis: verify generate-abis
```
Comment (technophile-04) [inline packages/foundry/Makefile:99]: Why do we need `generate-abis` here?

**Review by technophile-04 — COMMENTED** [review]
Hey thanks @portdeveloper! I think let's remove `--via-ir` flag from all the scripts. I think we don't need that (also not using in SE-2 hardhat) 

Also could you try verify `yarn verify --network sepolia` for me it's not working maybe it's not able to parse some json file correctly? It's because we removed `build-info` ? not sure could you check once? 

It seems to work in `foundry` branch. 

<details>

<summary>
    Here is example `YourContract.sol` so that it doesn't give already verified contract. 
</summary>

```solidity
//SPDX-License-Identifier: MIT
pragma solidity >=0.8.0 <0.9.0;

// Useful for debugging. Remove when deploying to a live network.
import "forge-std/console.sol";

/**
 * A smart contract that allows changing a state variable of the contract and tracking the changes
 * It also allows the owner to withdraw the Ether from the contract
 * @author BuidlGuidl
 */
contract YourContract {
    // State Variables
    address public immutable owner;
    string public greeting = "Building Unstoppable Apps!!!";
    uint256 public premiumThreshold = 0.01 ether;
    uint256 public totalCounter = 0;
    
    struct UserInfo {
        bool isPremium;
        uint256 greetingCounter;
        string[] greetingHistory;
        uint256 lastGreetingTime;
    }
    
    mapping(address => UserInfo) public userInfo;
    
    // Events
    event GreetingChange(
        address indexed greetingSetter, 
        string newGreeting, 
        bool premium, 
        uint256 value,
        uint256 timestamp
    );
    event PremiumThresholdUpdated(uint256 newThreshold);
    event PremiumStatusUpdated(address user, bool isPremium);

    // Constructor: Called once on contract deployment
    constructor(address _owner) {
        owner = _owner;
    }

    // Modifiers
    modifier isOwner() {
        require(msg.sender == owner, "Not the Owner");
        _;
    }

    modifier checkPremiumCooldown() {
        if (userInfo[msg.sender].isPremium) {
            require(
                block.timestamp >= userInfo[msg.sender].lastGreetingTime + 1 hours,
                "Premium users must wait 1 hour between greetings"
            );
        } else {
            require(
                block.timestamp >= userInfo[msg.sender].lastGreetingTime + 4 hours,
                "Regular users must wait 4 hours between greetings"
            );
        }
        _;
    }

    /**
     * Function that allows anyone to change the state variable "greeting" of the contract
     * Premium users get special benefits
     * @param _newGreeting (string memory) - new greeting to save on the contract
     */
    function setGreeting(string memory _newGreeting) public payable checkPremiumCooldown {
        require(bytes(_newGreeting).length > 0, "Greeting cannot be empty");
        require(bytes(_newGreeting).length <= 100, "Greeting too long");

        console.logString("Setting new greeting");
        console.logString(_newGreeting);

        // Update premium status if enough ETH is sent
        if (msg.value >= premiumThreshold) {
            userInfo[msg.sender].isPremium = true;
            emit PremiumStatusUpdated(msg.sender, true);
        }

        // Update greeting and counters
        greeting = _newGreeting;
        totalCounter += 1;
        userInfo[msg.sender].greetingCounter += 1;
        userInfo[msg.sender].lastGreetingTime = block.timestamp;

        // Store greeting history (only for premium users)
        if (userInfo[msg.sender].isPremium) {
            userInfo[msg.sender].greetingHistory.push(_newGreeting);
        }

        emit GreetingChange(
            msg.sender,
            _newGreeting,
            userInfo[msg.sender].isPremium,
            msg.value,
            block.timestamp
        );
    }

    /**
     * Function to update the premium threshold
     * @param _newThreshold - new minimum amount of ETH required for premium status
     */
    function updatePremiumThreshold(uint256 _newThreshold) public isOwner {
        require(_newThreshold > 0, "Threshold must be greater than 0");
        premiumThreshold = _newThreshold;
        emit PremiumThresholdUpdated(_newThreshold);
    }

    /**
     * Function to get a user's greeting history
     * @param _user - address of the user
     * @return string[] memory - array of user's previous greetings
     */
    function getGreetingHistory(address _user) public view returns (string[] memory) {
        require(userInfo[_user].isPremium, "User is not premium");
        return userInfo[_user].greetingHistory;
    }

    /**
     * Function that allows the owner to withdraw all the Ether in the contract
     */
    function withdraw() public isOwner {
        uint256 balance = address(this).balance;
        require(balance > 0, "No funds to withdraw");
        
        (bool success,) = owner.call{value: balance}("");
        require(success, "Failed to send Ether");
    }

    /**
     * Function that allows the contract to receive ETH
     */
    receive() external payable {}
}
```

</details>

**Comment by portdeveloper** [comment]
@technophile-04 

Verify worked for me!

https://github.com/user-attachments/assets/702aed03-5cae-4068-a9e9-93e28b2deaf7

**Review by technophile-04 — COMMENTED** [review]
> Verify worked for me! What was the error that you got?

Ohh it works now nicely for me too, I was actually getting the error in `VerifyAll.s.sol` that it was not able to parse json file. But I cleaned `broadcast`, `out` and `cache` dir and now it seems to work 🙌 

---

Just pushed a commit at https://github.com/scaffold-eth/scaffold-eth-2/pull/1020/commits/4d48b1e99cd7e8c791c17757bc3eee3529f3240f removing `deploy-and-verify`. I think it's better we keep the deployment, verify separate and also only have one way of verification(which is now similar to SE-2 hardhat) so that we don't confuse people also downside of having verification in same as deploy is that if verify fails it won't generate ts abis

Thanks @portdeveloper for catching it! Merging this 🙌

**Review by technophile-04 — APPROVED** [review]

**Pull request merged**